xapian-bindings/java/SmokeTest.java

```diff
@@ -121,8 +121,20 @@ public static void main(String[] args) throws Exception {
 		System.exit(1);
 	    }
 	    Enquire enq = new Enquire(db);
-	    enq.setQuery(new Query(Query.OP_OR, "there", "is"));
+	    enq.setQuery(new Query("dog"));
 	    MSet mset = enq.getMSet(0, 10);
+	    // MSet size is 0
+	    if (mset.size() != 0) {
+	    	for (MSetIterator.MatchDescriptor descriptor : mset) {
+	    		System.err.println("Empty iterator entered invalid section (from MSet java iterator)");
+	    		System.exit(1);
+	    	}
+	    } else {
+	    	System.err.println("Unexpected mset.size()");
```

Comment:
Since this matches an error further down, it'd be useful to distinguish them (perhaps "…, should be 0" here).